Intermittent crash on offline pages

I could not replicate that, but it happens sometimes

`01-20 10:49:38.270 11636 12279 E AndroidRuntime: FATAL EXCEPTION: AsyncTask #5
01-20 10:49:38.270 11636 12279 E AndroidRuntime: Process: com.brave.browser, PID: 11636
01-20 10:49:38.270 11636 12279 E AndroidRuntime: java.lang.RuntimeException: An error occurred while executing doInBackground()
01-20 10:49:38.270 11636 12279 E AndroidRuntime: 	at org.chromium.base.AsyncTask$NamedFutureTask.done(AsyncTask.java:833)
01-20 10:49:38.270 11636 12279 E AndroidRuntime: 	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
01-20 10:49:38.270 11636 12279 E AndroidRuntime: 	at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
01-20 10:49:38.270 11636 12279 E AndroidRuntime: 	at java.util.concurrent.FutureTask.run(FutureTask.java:271)
01-20 10:49:38.270 11636 12279 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
01-20 10:49:38.270 11636 12279 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
01-20 10:49:38.270 11636 12279 E AndroidRuntime: 	at java.lang.Thread.run(Thread.java:764)
01-20 10:49:38.270 11636 12279 E AndroidRuntime: Caused by: java.lang.IllegalArgumentException: Failed to find configured root that contains /storage/3205-026F/Android/data/com.brave.browser/files/Download/巴黎25 日天氣 - AccuWeather 巴黎法國預報 (ZH-TW).mhtml
01-20 10:49:38.270 11636 12279 E AndroidRuntime: 	at android.support.v4.a.d.a(FileProvider.java:719)
01-20 10:49:38.270 11636 12279 E AndroidRuntime: 	at android.support.v4.a.b.getUriForFile(FileProvider.java:404)
01-20 10:49:38.270 11636 12279 E AndroidRuntime: 	at org.chromium.chrome.browser.util.ChromeFileProvider.generateUri(ChromeFileProvider.java:74)
01-20 10:49:38.270 11636 12279 E AndroidRuntime: 	at org.chromium.chrome.browser.offlinepages.OfflinePageUtils$2.doInBackground(OfflinePageUtils.java:2540)
01-20 10:49:38.270 11636 12279 E AndroidRuntime: 	at org.chromium.base.AsyncTask$2.call(AsyncTask.java:436)
01-20 10:49:38.270 11636 12279 E AndroidRuntime: 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
01-20 10:49:38.270 11636 12279 E AndroidRuntime: 	... 3 more
01-20 10:49:38.275  1415  1803 E ActivityManager: App crashed! Process: com.brave.browser`

--- a/chrome/android/java/src/org/chromium/chrome/browser/offlinepages/OfflinePageUtils.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/offlinepages/OfflinePageUtils.java
@@ -580,7 +580,16 @@ public class OfflinePageUtils {
                     assert(isSchemeContentOrFile(uri));
                     return uri;
                 }
-                return ChromeFileProvider.generateUri(activity, offlinePageFile);
+                try {
+                    Uri uri = ChromeFileProvider.generateUri(activity, offlinePageFile);
+                    return uri;
+                } catch (IllegalArgumentException e) {
+                    assert false;
+                    if (pageUrl.isEmpty()) {
+                        return Uri.parse("https://");
+                    }
+                    return Uri.parse(pageUrl);
+                }
             }
             @Override
             protected void onPostExecute(Uri uri) {

